doc/tutorials/12-constant_pH/12-constant_pH.ipynb

```diff
@@ -166,130 +314,251 @@
-    "z = scipy.stats.norm.interval(1 - 0.05 / 2, loc=0)[1] # two-tail z-score at 95%\n",
-    "ci95 = z * np.array(std_dev_degree_of_dissociation) / np.sqrt(num_samples)\n",
-    "pK = -np.log10(K)\n",
+    "# plot the simulation results compared with the ideal results of the cations\n",
     "plt.figure(figsize=(10, 6), dpi=80)\n",
-    "plt.errorbar(pHs - pK, degrees_of_dissociation, ci95, label=r\"simulation\")\n",
-    "pHs2 = np.linspace(0, 8, num=50)\n",
-    "plt.plot(pHs2 - pK, ideal_degree_of_dissociation(pHs2, pK), label=r\"ideal\")\n",
-    "plt.xlabel('$pH-pK$', fontsize=20)\n",
-    "plt.ylabel(r'$\\alpha$', fontsize=20)\n",
-    "plt.legend(fontsize=20)\n",
+    "plt.errorbar(pHs,\n",
+    "             av_c.to('mol/L').magnitude,\n",
+    "             err_c.to('mol/L').magnitude,\n",
+    "             marker='o', c=\"tab:blue\", linestyle='none',\n",
+    "             label=r\"measured $c_{\\mathrm{B^+}}$\", zorder=2)\n",
+    "plt.plot(full_pH_range, c_H.to('mol/L').magnitude, c=\"tab:green\",\n",
+    "         label=r\"ideal $c_{\\mathrm{H^+}}$\", zorder=0)\n",
+    "plt.plot(full_pH_range, c_M.to('mol/L').magnitude, c=\"tab:orange\",\n",
+    "         label=r\"ideal $c_{\\mathrm{M^+}}$\", zorder=0)\n",
+    "plt.plot(full_pH_range, c_A.to('mol/L').magnitude, c=\"tab:blue\", ls=(0,(5,5)),\n",
+    "         label=r\"ideal $c_{\\mathrm{A^-}}$\", zorder=1)\n",
+    "plt.yscale(\"log\")\n",
+    "plt.ylim(1e-6,)\n",
+    "plt.xlabel('input pH', fontsize=16)\n",
+    "plt.ylabel(r'$c$ in $\\mathrm{mol/L}$', fontsize=16)\n",
+    "plt.legend(fontsize=16)\n",
     "plt.show()"
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "We now compare the relative frequencies $h$ of observed numbers of particles of type A to the probabilities $p$ which are expected in the constant pH ensemble for a non-interacting system [1]:\n",
+    "The plot shows that at intermediate pH the concentration of $\\mathrm{B^+}$ ions is approximately equal to the concentration of $\\mathrm{M^+}$ ions. Only at one specific $\\mathrm{pH}$ the concentration of $\\mathrm{B^+}$ ions is equal to the concentration of $\\mathrm{M^+}$ ions. This is the pH one obtains when dissolving the weak acid $\\mathrm{A}$ in pure water.\n",
     "\n",
-    "\\begin{equation}\n",
-    "    p(N_\\mathrm{A})=\\frac{{N_0 \\choose N_\\mathrm{A}} 10^{(\\mathrm{pH-p}K)N_\\mathrm{A}}}\n",
-    "        {\\sum_{N=0}^{N_0} {N_0 \\choose N} 10^{(\\mathrm{pH}-\\mathrm{p}K)N}}\n",
-    "\\end{equation}"
+    "In an ideal system, the ions missing in the simulation have no effect on the results. In an interacting system, the presence of ions in the box affects the properties of other parts of the system. Therefore, in an interacting system this discrepancy is harmless only at intermediate pH. The effect of the small ions on the rest of the system can be estimated from the overall the ionic strength.\n",
+    "$$ I = \\frac{1}{2}\\sum_i c_i z_i^2 $$"
```

Comment:
This is not quite true! It definitely depends on the observable you are looking at. In an ideal system the pressure is given by P=\sum_i c_i kT. If the concentration of some species i is wrongly represented, then the pressure which is observed is already wrong. Please adapt the sentence to reflect this fact.

Reply:
This is a more general issue, namely what is the meaning of "pressure" in implicit solvent simulations. One could claim that it is the osmotic pressure, but then one needs to specify what is on the other side of the osmotic membrane. This discussion is for sure outside the scope of the reaction toturial

I can address the comment by explicitly stating that the ions do not affect the ionization degree.

Reply:
The instantaneous "kinetic" virial pressure tensor component i,j is given by:
![image](https://user-images.githubusercontent.com/5189693/67199817-79acf200-f402-11e9-8ba0-f819a47461b8.png),
The last line is the ensemble averaged kinetic virial pressure tensor component i,j. Via some lines of math using the Maxwell-Boltzmann distribution of velocity components one obtains <P_i,j>=\sum_k c_k KT (where k runs over all ion species).

The instantaneous kinetic virial pressure tensor is returned by the pressure tensor command in Espresso.
The result of this observable (and also the pressure observable) will be affected if there are e.g. H+ ions missing at low pH in the cpH ensemble. Therefore the statement  "the ions missing in the simulation have no effect on the results [in an ideal system]" does not hold in general.